include/oneapi/dpl/internal/scan_by_segment_impl.h

```diff
@@ -0,0 +1,506 @@
+/*
+ *  Copyright (c) Intel Corporation
+ *
+ *  Copyright 2008-2013 NVIDIA Corporation
```

Comment:
Is it correct with legal perspective?

Reply:
The algorithm implemented is similar to the `reduce_by_key` and `scan_by_key` implementations in CUB and moderngpu. It is not a direct usage of their source code, but I included the NVIDIA copyright for this reason.